### ISSUE TITLE
App Insights Result Code showing "0"

### ISSUE THREAD

**Issue by ketaanhshah** [microsoft/ApplicationInsights-node.js#553]
Hi All,
We have an application which makes AJAX calls and stuff is reported back to App Insights. We can understand when the Result code returns 500 error code due to Server Failure, but some records show us FAILURE but the Status code is 0 and we are wondering what could be the case any help on this? Following is some sample data for same call being made and failures happen but some status codes are 500 and some are 0. Any help how we can find out whats happening and what is the meaning of "0" status code:

timestamp | id | target | type | name | data | success | resultCode | duration | performanceBucket
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
2019-09-19T19:24:34.418Z | \|3GBot.aBg6B | quote.xxxxxxxxxxxxxx.com \| cid-v1:ac4be586-8386-4f62-baf2-be19388b8d61 | Ajax | POST /quote/ReCacheUpdatedData | POST /quote/ReCacheUpdatedData | FALSE | 500 | 1320 | 1sec-3sec
2019-09-19T19:24:34.418Z | \|3GBot.aBg6B | quote.xxxxxxxxxxxxxx.com \| cid-v1:ac4be586-8386-4f62-baf2-be19388b8d61 | Ajax | POST /quote/ReCacheUpdatedData | POST /quote/ReCacheUpdatedData | FALSE | 500 | 1320 | 1sec-3sec
2019-09-23T20:39:44.12Z | \|B2MWb.D0ayE | quote.xxxxxxxxxxxxxx.com | Ajax | POST /quote/ReCacheUpdatedData | POST /quote/ReCacheUpdatedData | FALSE | 0 | 1416 | 1sec-3sec
2019-09-23T20:59:00.722Z | \|X8TMF.quPie | quote.xxxxxxxxxxxxxx.com | Ajax | POST /quote/ReCacheUpdatedData | POST /quote/ReCacheUpdatedData | FALSE | 0 | 6004 | 3sec-7sec

**Comment by markwolff** [comment]
It can mean several things, but is ultimately used as an "undefined" status code. Some scenarios: you are offline or no response was received.

**Comment by oviliz** [comment]
> It can mean several things...
This is Microsoft for you...

**Comment by markwolff** [comment]
https://stackoverflow.com/questions/872206/what-does-it-mean-when-an-http-request-returns-status-code-0/14507670#:~:text=An%20HTTP%20response%20code%20of,execute%20within%20a%20specified%20time.

**Comment by ajnavsun1806** [comment]
@ketaanhshah We also faced this issue recently and I have reproduced this issue in my dev machine. To reproduce this issue, I created a request from postman and put a stopwatch in my code. I cancelled the request from postman and then check the AppRequest trace in console. It contains result code as 0. But I am not sure about the solution for this issue. So probably one idea would be to add a processor and in case result code is 0 change result code to 500
